torch/csrc/autograd/custom_function.cpp

```diff
@@ -26,6 +29,136 @@ Variable VariableInfo::zeros(at::OptionalDeviceGuard& device_guard) const {
+    const auto& alias_info = arguments[i].alias_info();
+    if (alias_info.has_value() && alias_info->isWrite()) {
+      check_inplace(t, any_requires_grad);
+    }
+  }, stack, stack_start, num_arguments);
+
+  std::shared_ptr<NotImplemented> grad_fn;
+  if (any_requires_grad) {
+    grad_fn = std::shared_ptr<NotImplemented>(new NotImplemented(op_name), deleteNode);
+    grad_fn->set_next_edges(collect_next_edges(tensors_requiring_grad));
+  }
+
+  #ifndef NDEBUG
+  // See NOTE [ TensorImpl and Storage Pointer Sanity Checks ]
+  std::vector<c10::IValue> stack_copy(*stack);
+  std::vector<c10::intrusive_ptr<TensorImpl>> impl_saved;
+  impl_saved.reserve(num_tensor_inputs);
+  std::vector<c10::optional<Storage>> storage_saved;
+  storage_saved.reserve(num_tensor_inputs);
+  _foreach_tensor([&](size_t idx, size_t _, const at::Tensor& t) {
+    storage_saved.push_back(t.has_storage() ? c10::optional<Storage>(t.storage()) : c10::nullopt);
+  }, &stack_copy, stack_start, num_arguments);
+  _foreach_tensor([&](size_t idx, size_t _, const at::Tensor& t) {
+    impl_saved.push_back(t.getIntrusivePtr());
+  }, &stack_copy, stack_start, num_arguments);
+  #endif
+  {
+    at::AutoDispatchBelowADInplaceOrView guard;
+    op.redispatchBoxed(dispatch_keys & c10::after_autograd_keyset, stack);
+  }
+  #ifndef NDEBUG
+  _foreach_tensor([&](size_t idx, size_t _, const at::Tensor& t) {
+    if (storage_saved.at(idx).has_value())
+      AT_ASSERT(storage_saved.at(idx).value().is_alias_of(t.storage()));
+  }, &stack_copy, stack_start, num_arguments);
+  _foreach_tensor([&](size_t idx, size_t _, const at::Tensor& t) {
+    if (impl_saved.at(idx))
+      AT_ASSERT(impl_saved.at(idx) == t.getIntrusivePtr());
+  }, &stack_copy, stack_start, num_arguments);
+  // Do we have alias information for tensors in tensorlist outputs?
+  _foreach_tensor([&](size_t idx_tensor, size_t idx_ret, const at::Tensor& t) {
+    if (!is_inplace_output[idx_ret])
+      AT_ASSERT(t.use_count() <= 1);  // Okay to return undefined tensor
+  }, stack, stack->size() - num_returns, num_returns);
+  // TODO: Add check if view ops return tensor that is aliased with the right input
+  _foreach_tensor([&](size_t idx_tensor, size_t idx_ret, const at::Tensor& t) {
+    if (!is_aliased_output[idx_ret] && t.has_storage())
+      AT_ASSERT(t.storage().use_count() == 1);
+  }, stack, stack->size() - num_returns, num_returns);
+  #endif
+  const auto& ret = torch::jit::last(stack, num_returns);
+
+  if (any_requires_grad) {
+    for (const auto return_idx : c10::irange(ret.size())) {
+      if (ret[return_idx].isTensor()) {
+        const auto& t = ret[return_idx].toTensor();
+        if (isDifferentiableType(t.scalar_type())) {
+          if (is_inplace_output[return_idx]) {
+            rebase_history(const_cast<at::Tensor&>(t), grad_fn);
+          } else {
+            set_history(const_cast<at::Tensor&>(t), grad_fn);
+          }
+        }
+      }
```

Comment:
@albanD what about lists here?

Reply:
The same thing should be done.
In the codegen, we use `set_history(flatten_tensor_args( result ), grad_fn);` where result can be either a Tensor, a tuple or a vector<Tensor>.
But I don't think it handles `isDifferentiableType()` which is handled before.

So here we need to add a new else for isTensorList indeed